image_explorer/public/js/image_explorer.js

```diff
@@ -121,4 +121,22 @@ function ImageExplorerBlock(runtime, element) {
     publish_event({
         event_type:'xblock.image-explorer.loaded'
     });
+
+
+    // Responsive hotspots for mobiles i-e (< 767px)
+    $(window).load(function () {
+        $('.image-explorer-wrapper').each(function (index) {
+            var $imgWidth = $(this).find('img')[index].naturalWidth;
+            var $imgHeight = $(this).find('img')[index].naturalHeight;
+            $(this).find('.image-explorer-hotspot').each(function () {
+                var setLeft = $(this).position().left / $imgWidth * 100;
+                var setTop = $(this).position().top / $imgHeight * 100;
+                $(this).css({
+                    'left': setLeft + '%',
+                    'top': setTop + '%',
+                    'visibility': 'visible'
+                });
+            });
+        });
+    });
```

Comment:
I'm not a big fan of doing this conversion in JavaScript, because it causes things to move around on the mobile view after the page loads, and it's less dependable. What do you think about just asking authors to start using % for the hotspot positions in mobile courses?

1. Allow authors to put things like `x="50%"` right in the XML.
1. Change [the context](https://github.com/edx-solutions/xblock-image-explorer/blob/65b9503/image_explorer/image_explorer.py#L285-L286) to
    ```
    hotspot.x = hotspot_element.get('x')
    if not hotspot.x.endswith('%')`:
        hotspot.x += 'px'  # px is deprecated as it is not responsive
    ```
1. Change [the template](https://github.com/edx-solutions/xblock-image-explorer/blob/2a5aa75/image_explorer/templates/html/image_explorer.html#L12) from `{{hotspot.y}}px` to `{{hotspot.y}}`

Reply:
This is exactly the same approach that came in my mind but it will distort all the image explorers on production(as their values are in pixels) as we are merging it in master. 

First we need to make a request to change all the existing hotspots to % then we can merge it with master.

Thoughts? @zvishal  @wajeeha-khalid

Reply:
@murad-hubib  What I meant is to make it so that authors can specify **either** a '%' or a 'px' value - that way, we don't have to change any courses in production, but we can use '%' in all the new courses.

This means that the existing courses wouldn't be responsive, but that seems to be OK.